causync.py

```diff
@@ -22,51 +22,77 @@ class CauSync(object):
-            dst_abs (string): absolute path of destination directory, joined with date_ival
+            src_abs (list): list containing source directory absolute paths
+            dst_abs (str): absolute path of destination directory, joined with date_ival
+            curdate (datetime): datetime object containing the current date
             logger (object): the logger object used for logging to file/console
     """
 
     curdate = None
+    multiple_sources = False
 
-    def __init__(self, config, src, dst, task, no_incremental=False, quiet=False, dry_run=False, selfname="causync.py"):
+    def __init__(self, config, src, dst, task, no_incremental=False, quiet=False,
+                 dry_run=False, selfname="causync.py", excludes=None, exclude_from=False,
+                 loglevel=None):
         self.config = config
         self.name = selfname
         self.pid = os.getpid()
         self.no_incremental = no_incremental
+        self.excludes = excludes if excludes else []
+        if exclude_from:
+            self.excludes = self.excludes + self.parse_exclude_file(exclude_from)
 
         self.src = src
-        self.src_abs = os.path.abspath(self.src)
+        self.src_abs = self.parse_src(self.src)
+        if not self.src_abs:
+            exit(-1)
+
         self.dst = dst
         self.dst_abs = os.path.abspath(dst)
 
         self.task = task
         self.quiet = quiet
         self.dry_run = dry_run
         self.curdate = datetime.now()
-        self.logger = self.get_logger()
+        self.logger = self.get_logger(loglevel)
+
+    def parse_src(self, src):
+        if type(src) == type(list()):
+            self.multiple_sources = True if len(src) > 1 else False
+            return [os.path.abspath(i) for i in src]
+        elif type(self.src) == str:
+            return [os.path.abspath(self.src)]
+        else:
+            self.logger.error("source directory type error")
+            return False
 
     def run(self):
         """ Main run function. """
 
         is_running = self.is_running()
-        lockfile_exists = self.lockfile_exists()
+        lockfile_exists = self.check_lockfiles()
 
         self.logger.info("started with PID {}".format(self.pid))
+        if self.multiple_sources:
+            self.logger.info("syncing multiple source directories")
 
         if self.dry_run:
             self.logger.info("doing dry run")
         if self.task == 'cleanup':
             self.run_cleanup()
         elif self.task in ['check', 'sync']:
             if lockfile_exists or is_running:
-                self.logger.info("causync is already running on {}".format(self.src_abs))
+                self.logger.info("causync is already running on {}".format(", ".join(self.src_abs)))
             else:
-                self.logger.info("causync is not already running on {}".format(self.src_abs))
+                self.logger.info("causync is not already running on {}".format(", ".join(self.src_abs)))
```

Comment:
The "is not already running" is incorrect in English. Please use "is not running yet".